src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -321,6 +322,29 @@ private static bool IsInVarContext(NameSyntax name)
             return false;
         }
 
+        private static bool TryClassifyMethodGroupAssignment(
+            NameSyntax name,
+            SymbolInfo symbolInfo,
+            ArrayBuilder<ClassifiedSpan> result)
+        {
+            // If the syntax matches "var x = m" or "x = m", and there is a candidate symbol
+            // for m that is a method then classify it as such.
+            if (name is IdentifierNameSyntax identifierName &&
+                symbolInfo.Symbol == null &&
+                symbolInfo.CandidateSymbols.Length == 1 &&
+                symbolInfo.CandidateReason == CandidateReason.OverloadResolutionFailure &&
```

Comment:
Should we just count overload resolution as another case that sends us into the ambiguous path here?

https://github.com/dotnet/roslyn/blob/5256b889b47f5e120ad691d588c8c8b24351fe16/src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs#L72-L73

Reply:
Thanks for the suggestion, I will give it a try and see what breaks.

There is another spot where I thought a fix could go, where it allows overload resolution to still work for constructors, and I don't know why that wouldn't be the case for any other method, but I always assume that existing behaviour is there for a reason.

This PR is essentially a "is this fix reasonable?" question, because I didn't get an answer on Teams :)